src/core/models/tabsmodel.cpp

```diff
@@ -217,15 +228,95 @@ void TabsModel::setInactive()
 
 bool TabsModel::setPreviousTabActive()
 {
+    return moveByHistory(1);
+}
+
+bool TabsModel::setNextTabActive()
+{
+    return moveByHistory(-1);
+}
+
+bool TabsModel::activateTabRelativeToCurrent(int offset)
+{
+    if (m_tabs_list.length() == 0) {
+        activeChanged(m_active_tab = m_invalid_tab);
+        activeIndexChanged(activeIndex());
+        return false;
+    }
+
+    int currIndex = activeIndex();
+    if (currIndex == -1)
+    {
+        setActive(m_tabs_list.last(), false);
+        return true;
+    }
+
+    int newIndex = currIndex + offset;
+    if (newIndex < 0 || newIndex >= m_tabs_list.length())
+    {
+        return false;
+    }
+
+    return setActive(m_tabs_list.at(newIndex), false);
+}
+
+bool TabsModel::moveByHistory(int offset)
+{
     if (m_tabs_list.length() == 0) {
         activeChanged(m_active_tab = m_invalid_tab);
         activeIndexChanged(activeIndex());
         return false;
     }
     if (m_active_tab_history.length() == 0) {
-        setActive(m_tabs_list.first());
+        setActive(m_tabs_list.first(), false);
```

Comment:
In case the tab history is empty, I guess the first tab should be set active as fallback, or this line should be removed.